resmio_conversion.js

```diff
@@ -0,0 +1,195 @@
+/*!
+ * ===========================================
+ * General Conversion Tracking v1.0
+ * Copyright 2017 resmio GmbH (Philipp Sahner)
+ * ===========================================
+ */
++function ($) {
+  /*
+   * +++++++++++++++++++++++++++++++++++++++++++++++++++++
+   * Check the url for parameters when document has loaded
+   * and handle it accordingly
+   * +++++++++++++++++++++++++++++++++++++++++++++++++++++
+   */
+  $(document).ready(function(){
+    // get the url, check it for parameters and update the links
+ 	  var url = window.location.href,
+        obj = getAllUrlParams(url);
+        isGclid = false;
+        isObj = false;
+    if (!($.isEmptyObject(obj))) {
+      isObj = true;
+      // create the cookies for the parameters
+      $.each( obj, function( key, value ) {
+        var current_value = readCookie(key),
+            value = value.toString().toLowerCase();
+        if (value != current_value) {
+          createCookie(key, value, 90);
+        }
+        if (key == 'gclid') {
+          isGclid = true;
+        }
+      });
+    }
+    // no gclid parameter but gclid cookie
+    // than add it to the parameters/links
+    if (isGclid == false) {
+      // check cookies for gclid and add it to object if it is there
+      var gclidValue = readCookie('gclid');
+      if (gclidValue != '') {
+        // obj['gclid'].push(gclidValue);
+        obj['gclid'] = gclidValue;
+      }
+    }
+    updateLinks(url, obj);
+  });
+
+  /*
+   * +++++++++++++++++
+   * General functions
+   * +++++++++++++++++
+   */
+
+  /*
+   * Get all the parameters of an URL as object (array)
```

Comment:
it is either an object or an array (although technically an array is an object inside the JS engine, but we don't want to scuba dive yet ;) )